src/typval.c

```diff
@@ -1760,7 +1760,7 @@ typval_compare_object(
 
     for (int i = 0; i < cl1->class_obj_member_count; ++i)
 	if (!tv_equal((typval_T *)(obj1 + 1) + i,
-				 (typval_T *)(obj2 + 1) + i, ic, TRUE))
```

Comment:
@LemonBoy  The current version already has tests about object(enum) equality.

https://github.com/vim/vim/blob/7002c055d560ae0b3bb1e24ad409390a5443daad/src/testdir/test_vim9_enum.vim#L705-L725

These tests will fail, we still assign TRUE to the variable `recursive`.

Reply:
Not with the patch I've suggested, the test suite runs just fine here.